test/uk/gov/hmrc/cataloguefrontend/createarepository/CreateRepoFormSpec.scala

```diff
@@ -0,0 +1,256 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.cataloguefrontend.createarepository
+
+import uk.gov.hmrc.cataloguefrontend.util.UnitSpec
+
+class CreateRepoFormTest extends UnitSpec {
+
+  "repoNameWhiteSpaceValidation" when {
+    "The repo name contains a space" should {
+      "return false" in {
+        CreateRepoForm.repoNameWhiteSpaceValidation("hello world") shouldBe false
+        CreateRepoForm.repoNameWhiteSpaceValidation(" helloworld") shouldBe false
+        CreateRepoForm.repoNameWhiteSpaceValidation("helloworld ") shouldBe false
+      }
+    }
+
+    "The repo name contains multiple spaces" should {
+      "return false" in {
+        CreateRepoForm.repoNameWhiteSpaceValidation("   hello world") shouldBe false
+        CreateRepoForm.repoNameWhiteSpaceValidation("hello         world") shouldBe false
+        CreateRepoForm.repoNameWhiteSpaceValidation("helloworld   ") shouldBe false
+      }
+    }
+
+    "The repo name contains a tab" should {
+      "return false" in {
+        CreateRepoForm.repoNameWhiteSpaceValidation("\thello world") shouldBe false
+      }
+    }
+
+    "The repo name contains a newline" should {
+      "return false" in {
+        CreateRepoForm.repoNameWhiteSpaceValidation("\nhello world") shouldBe false
+      }
+    }
+
+    "The repo name contains no whitespace characters" should {
+      "return true" in {
+        CreateRepoForm.repoNameWhiteSpaceValidation("helloworld") shouldBe true
+      }
+    }
+  }
+
+  "repoNameLengthValidation" when {
+    "the repo name contains less than 48 characters" should {
+      "return false" in {
+        CreateRepoForm.repoNameLengthValidation("helloworld") shouldBe true
+      }
+    }
+
+    "the repo name contains exactly 47 characters" should {
+      "return true" in {
+        CreateRepoForm.repoNameLengthValidation("helloworldhelloworldhelloworldhelloworldhellowo") shouldBe true
```

Comment:
You could do something like `LazyList.continually("a").take(47).mkString` to avoid handcrafting strings of particular lengths